rulesets/nixpkgs/no-creation.json

```
@@ -7,9 +7,17 @@
   "enforcement": "active",
   "conditions": {
     "ref_name": {
-      "exclude": [],
+      "exclude": [
+        "refs/heads/backport-*",
+        "refs/heads/nixos-*",
+        "refs/heads/nixpkgs-*",
+        "refs/heads/release-*",
+        "refs/heads/revert-**",
```

Should this be

```suggestion
        "refs/heads/revert-*",    
```

Or is there no difference?

---

There is a difference: `**` will include `/`. Aka when you have a branch called `revert-123456-feat/stuff`, `revert-*` will not match, but `revert-**` will.

Since the `revert-` branches are auto-created from the PR number *and the head branch of the PR*, which was provided by a regular contributor, this can use the `/` format.

I'm OK with not allowing these for `wip-` and `backport-` prefixes, but for the above reason we must support them for `revert-`.

Edit: This is a moment where I miss the ability to add comments to `.json`...

---

> There is a difference: `**` will include `/`. Aka when you have a branch called `revert-123456-feat/stuff`, `revert-*` will not match, but `revert-**` will.

That's what the docs tell me, at least: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/creating-rulesets-for-a-repository#using-fnmatch-syntax

But it's apparently not that easy, because @philiptaron [just hit this](https://github.com/NixOS/nixpkgs/pull/420319#issuecomment-3079769010) when trying to revert a r-ryantm. The branch is called `auto-update/re2` - and is blocked by our new rule.

I reproduced this in my fork as well. Regular branches work well, but those with `/` don't.